src/Project.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+namespace Storj\Uplink;
+
+use DateTimeImmutable;
+use FFI;
+use FFI\CData;
+use Generator;
+use Storj\Uplink\Exception\UplinkException;
+use Storj\Uplink\Internal\Scope;
+use Storj\Uplink\Internal\Util;
+
+class Project
+{
+    /**
+     * With libuplink.so and header files loaded
+     */
+    private FFI $ffi;
+
+    /**
+     * The associated C struct of type Project
+     */
+    private CData $cProject;
+
+    /**
+     * To free memory when this object is done
+     */
+    private Scope $scope;
+
+    public function __construct(FFI $ffi, CData $cProject, Scope $scope)
```

Comment:
Do we really need to have this constructor public?

Reply:
It needs to be public to construct it from class Access.

There's no package or module visibility like Java.